src/runtime/BorrowedReference.cs

```diff
@@ -11,8 +11,7 @@ namespace Python.Runtime
         public bool IsNull => this.pointer == IntPtr.Zero;
 
         /// <summary>Gets a raw pointer to the Python object</summary>
-        public IntPtr DangerousGetAddress()
-            => this.IsNull ? throw new NullReferenceException() : this.pointer;
+        public IntPtr DangerousGetAddress() => this.pointer;
```

Comment:
This should not be changed. If a function is needed, that allows access to the raw pointer when it is `null`, it should be a separate function for clarity.

Reply:
I'll add an unchecked version

Reply:
That checking is not necessary, because the method already named DangerousXXX, why it even can't obtain a null pointer. If it needs some null validation, it should do by the invoker not the internal.

Reply:
@amos402 if the check is missed by accident by the invoker, the bug would be much easier to trace if it fails at the first access rather than much later on, when the raw null pointer is actually dereferenced.

Reply:
What I concern is the method conformed the single responsibility or not. It is unfit that a method like this can't obtain a null pointer. For example, `std::shared_ptr<T>::get` can illustrate this point.
Anyway, I don't want to argue about this, if you insist, just check it, I don't mind it.

Reply:
I tend to agree with @amos402, a method prefixed with `Dangerous` should let me do dangerous things. However, I think this modification should go in another PR as it modifies the current behaviour, and adding a checked version would add up a lot of changes all over the place.